assayist/processor/container_go_analyzer.py

```diff
@@ -105,21 +110,87 @@ def run(self):
             raise AnalysisFailure('GoAnalyzer completed with the following error(s): \n  {}'
                                   .format("\n  ".join(failures)))
 
-    def _process_source_code(self, source_location, srcdir, excludes=None):
-        """Run backvendor on the source code.
+    def _process_git_source(self, source_location, srcdir):
+        """Run backvendor on the dist-git repository source code.
+
+        Exclude files known to be specific to the git repository and
+        not expected to be upstream.
+
+        Parse the output and update the database accordingly.
 
         :param SourceLocation source_location: local source code DB node
-        :param str path: path to source code to examine
-        :param list excludes: list of globs to ignore
+        :param str srcdir: path to source code to examine
+        """
+        # In case we need to provide an import path, see if there is
+        # an io.openshift.source-repo-url label in the Dockerfile to help
+        # us guess one.
+        import_path_from_source_url = self._get_import_path_override(srcdir)
+
+        # If we have a reasonable guess, see if we will need to use it.
+        import_path = None
+        if (import_path_from_source_url and
+                not self._import_paths_known(srcdir,
+                                             excludes=self.DIST_GIT_EXCLUDES)):
```

Comment:
@twaugh, the last important think is that `backendor` does not  fail in `_import_paths_known`
and thus RuntimeExceptions is not raised. So result of this condition will be false.

Reply:
In that situation, backvendor really did succeed -- it knows the top-level import path, and that's all it was asked for.

The case where `-importpath` will help *does* fail:
```
$ mkdir /tmp/x
$ cd /tmp/x
$ echo package x > x.go
$ backvendor -x -debug -only-importpath .
2018/11/29 14:35:52 found project at top-level: .
2018/11/29 14:35:52 .: unable to determine import path
Provide import path with -importpath
$ echo $?
1
```

Reply:
Then I would probably need to check with different image which has such label